### PR TITLE
PWN-5528 - add transaction for free renBTc

### PR DESCRIPTION
**Pull request by davran1001**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5528

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/rpc/RpcModule.kt [hunk 1]
```diff
@@ -77,7 +77,7 @@ object RpcModule : InjectionModule {
         factory {
             TransactionInteractor(get(), get(), get(), get())
         }
-        factory { TokenInteractor(get(), get(), get(), get()) }
+        factory { TokenInteractor(get(), get(), get(), get(), get(), get()) }
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/rpc/RpcModule.kt:80]: factoryOf ?

**Review by kabramovp2p — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/feerelayer/interactor/FeeRelayerInteractor.kt [hunk 2]
```diff
@@ -220,4 +220,25 @@ class FeeRelayerInteractor(
...
+        if (!feePayer.equals(preparedTransaction.transaction.feePayer)) {
+            throw IllegalStateException("Invalid fee payer")
+        }
+
+        /*
+        * Retrying 3 times to avoid some errors
+        * For example: fee relayer balance is not updated yet and request will fail with insufficient balance error
+        * */
+        return retryRequest {
+            feeRelayerRepository.relayTransaction(preparedTransaction.transaction, statistics).firstOrNull().orEmpty()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/feerelayer/interactor/FeeRelayerInteractor.kt:241]: chain like formatting please

Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/feerelayer/interactor/FeeRelayerInteractor.kt:241]:
```suggestion
            feeRelayerRepository.relayTransaction(preparedTransaction.transaction, statistics)
                .firstOrNull()
                .orEmpty()
```

File: app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt [hunk 3]
```diff
@@ -36,7 +46,17 @@ class TokenInteractor(
         return rpcRepository.sendTransaction(transaction)
     }
 
-    suspend fun openTokenAccount(
+    suspend fun createAccount(mintAddress: String): String {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt:49]: you can use Base58String to specify string format

File: app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt [hunk 4]
```diff
@@ -36,7 +46,17 @@ class TokenInteractor(
...
+        val signUpDetails = userSignUpDetailsStorage.getLastSignUpUserDetails()?.signUpDetails
+        val isWeb3AuthUser = signUpDetails?.ethereumPublicKey != null
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt:51]: i check isWeb3AuthUser just using `userSignUpDetailsStorage.getLastSignUpUserDetails() != null`

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**